src/xg.hpp

```diff
@@ -99,6 +99,13 @@ class XG {
     size_t node_count;
     size_t edge_count;
     size_t path_count;
+    
+    // We need the w function, which we call the "where_to" function. It tells
+    // you, from a given visit at a given side, what visit offset if you go to
+    // another side.
+    int64_t where_to(int64_t current_side, int64_t visit_offset, int64_t new_side) const;
+    int64_t where_to(int64_t current_side, int64_t visit_offset, int64_t new_side,
+      vector<Edge>& edges, vector<Edge>& edges_out) const;
```

Comment:
What do the new parameters mean? Is the caller supposed to look at them or just preserve them across multiple calls? Are the supposed to be specific to the side being asked about, or shared between multiple sides?